### PR TITLE
first version lightning page

### PR DESCRIPTION
**Pull request by amberrignell**
A whole page to tell people about Lightning ! Hopefully we can send this as a resource to people we're introducing to Lightning. Let me know what you think and if you have any opinions on what the Q&A order should be (if it matters at all). I've ported the text from 'The integration toolkit' page over too.

I'll work on a Features and Roadmap section next week.

### CODE REVIEW

**Comment by taylordowns2000** [comment]
@amberrignell , is this ready for review? (convert from draft if so please!)
image

File: docs/about-lightning.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+title: Lightning (Coming soon)
+sidebar_label: Lightning
+---
+
+## The Lightning pitch
+
+##### Leveraging the tech powering the field-tested enterprise OpenFn platform...
+
+Lightning brings together the tried and tested technology we have been using
+since 2015 (OpenFn core, engine, adaptors and code from the OpenFn SaaS
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:11]: Can we just say `the OpenFn Integration Toolkit Digital Public Good` instead of listing the components only our team knows about? `(OpenFn core, engine, adaptors and code from the OpenFn SaaS platform)`

File: docs/about-lightning.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
+---
+title: Lightning (Coming soon)
+sidebar_label: Lightning
+---
+
+## The Lightning pitch
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:6]: Can we please have a different title than `The Lightning pitch`? This sounds like it's copied from internal draft documents ... maybe `Introducing Lightning`

File: docs/about-lightning.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+scalable and secure way.
+
+##### ...but providing a fully open source web app with a user-friendly web interface.
+
+A fully open source web app, it can be deployed anywhere through Docker and
+comes with a user-friendly, low-code interface with the full functionality
+needed for organizations to build, run and audit their workflows all in one
+place. ![Lightning](/img/lightning_build_run_audit.png)
+
+##### Build
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:22]: If we have them, it would be great to add screesnhots for the `Build` and `Audit` interfaces to people can visual what's coming

File: docs/about-lightning.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+##### Audit
+
+Treat every workflow run with the care and attention it deserves. In OpenFn,
+each incoming request or transaction that gets processed is more than a piece of
+data - it represents a vulnerable child in need of critical support, a farmer
+managing their savings to make sure they can afford the next harvest. Lightning
+provides users with a dashboard that allows them to monitor the health of their
+integrations to make sure no request goes unprocessed.
+
+## What is Lightning ?
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:39]: We've already said (in a different way) what Lightning is above. Let's explain "what is Lightning" one time, and then use subsequent sections to dig more into details... so maybe here in this section we want to talk more specifically about `Lightning as a Workflow Engine` or something like that. 
But this section does not flow with the above, seems redundant.

File: docs/about-lightning.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+Lightning is developed in line with the
+[principles for digital development](https://digitalprinciples.org/principles/)
+and under the guidance of it's Open Source Steering Committee which you can read
+about [here](https://openfn.github.io/governance/OSSC.html).
+
+On top of this, Lightning follows 4 key principles which determine how it should
+be developed:
+
+### Integrations break
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:64]: Maybe instead of having a header `Integrations break` (a line I don't want associated with Lightning), what about something like `Interoperability is an ongoing process` instead of the current title?

File: docs/about-lightning.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+Anyone that has worked on integration projects in the past is well aware that
+integrations do break. No matter how well designed they are, the fact is that
+**they connect multiple systems that all change over time**: new API versions
+get released, data models change, IDs, codes and mappings change, data standards
+are updated and the processes themselves evolve. This is why Lightning will
+include:
+
+Enhanced testing and debugging:
+
+- Save inputs as edge cases for robust workflow testing
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:75]: What is an input? ...Test data?

File: docs/about-lightning.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+- View the input and output for each step in a workflow to easily identify where
+  an error occurred
+- Throw custom errors to improve API messages (adaptors)
+- Add custom logic to handle a step failure (fail triggers)
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:79]: Change `step` to `workflow step` or `process step`? (Otherwise with little context, people might not know what a `step` is

File: docs/about-lightning.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+
+First of class monitoring:
+
+- Get notified on run failures
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:83]: What is a `run`? Remember, people may not know how OpenFn works

File: docs/about-lightning.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+- View the status of every run
+- Search workflow runs by input/output data and logs
+- Filter workflow runs by status, workflow name and date
+
+Re-processing functionality:
+
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:89]: `Bulk reprocess requests`... what is a `request` in this context? Let's be more specific for people who don't know... maybe instead use `API requests` or `workflow runs`?

File: docs/about-lightning.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+developers collaborate**. Lightning is being developed to bridge the gap between
+non-technical and technical users through:
+
+Intuitive, user-friendly user interface for non-technical users:
+
+- Understand a workflow in a visual, human-readable format (abstract away from
+  code to make workflows understandable to non-technical users)
+- Build credentials through a form (remove the need to read through confusing
+  API documentation)
+- Build requests through a form
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:114]: Again reading this as I don't know anything about OpenFn... 
What are `requests`? HTTP requests? 
And "through a form" ... `through a form _interface_` maybe?

File: docs/about-lightning.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+- Save mappings used in workflows as constants so they can be easily viewed and
+  edited without needing to read code
+- Clear documentation for users to learn how to plan and build integrations
+
+Projects-as-code and CLI for a developer interface:
+
+- Export, import and configure projects as code in the code editor of your
+  choice
+- Run, test and deploy projects through a command line interface
+- Review and track changes through github version control
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:124]: It doesn't have to be only Github version control, right?

File: docs/about-lightning.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+
+Security:
+
+- Secure credential management (encrypted at REST, credential secrets are
+  scrubbed from logs, secure credential sharing
+- Zero-persistence pipelines
+- Role-based project access
+- Additional authentication rules for webhooks
+
+### Standards and compliance matter
```
Comment (aleksa-krolls) [inline docs/about-lightning.md:156]: Standards is a major selling point... and this is the first time we mention "Digital Public Good"! I think this should maybe be moved to the top.

File: docs/getting-started/integration-toolkit.md [hunk 13]
```diff
@@ -13,59 +13,7 @@ in integration and interoperability projects. The Toolkit is both a recognized
 The Toolkit provides a suite of software tools and documentation to help users
 design, build, and automate integrations.
 
-## Lightning Coming Soon!
```
Comment (aleksa-krolls) [inline docs/getting-started/integration-toolkit.md:16]: I do not think you should remove all mentions of `Lightning` on this page... if you do, Lightning does not show up anywhere here and is not on the diagram screenshot on this page. @amberrignell 

So either (1) update the Toolkit diagram to include Lightning, and/or (2) keep a small section mentioning Lightning as the latest and greatest extension of the DPG... and then maybe link to your main Lightning page.

**Review by aleksa-krolls — CHANGES_REQUESTED** [review]
@amberrignell I've left comments throughout. If you want me to merge this PR just so we can get your page live ASAP, I'm happy to do that so you have something to link to... so long as my comments are addressed sometime this week. 

Otherwise if you have time, pls address my comments before we merge. 

Keep me posted on how you want to proceed.

File: docs/about-lightning.md (hunk 5 above)
Comment (amberrignell) [reply docs/about-lightning.md:64]: Happy with this for now but would be good to think of something a bit shorter/more memorable down the line

File: docs/about-lightning.md (hunk 6 above)
Comment (amberrignell) [reply docs/about-lightning.md:75]: how about this: Save data from workflow runs as test data for robust workflow testing of edge cases

File: docs/about-lightning.md (hunk 9 above)
Comment (amberrignell) [reply docs/about-lightning.md:89]: Bulk reprocess workflow runs after updating workflow steps to course-correct if a workflow has been running with flawed logic ?

**Comment by aleksa-krolls** [comment]
@amberrignell as discussed, this is merged for now - but please revisit my comments later
